### ISSUE TITLE
TRUNCATE parent_table CASCADE doesn't purge child_table chunks

### ISSUE THREAD

**Issue by jflambert** [timescale/timescaledb#2109]
**Relevant system information:**
 - OS: Ubuntu 18.04
 - PostgreSQL version (output of `postgres --version`): 12.3
 - TimescaleDB version (output of `\dx` in `psql`): 1.7.2
 - Installation method: Docker

**Describe the bug**
Truncating a "parent table" with CASCADE does purge data from the "child table" but it does not purge the underlying chunks. Truncating the "child table" (at that point empty) does get rid of the chunks.

**To Reproduce**
1. Create the structure and some data. This will create 6 chunks.

```
DROP TABLE IF EXISTS parent_table CASCADE;
CREATE TABLE parent_table
(
  p_id INT PRIMARY KEY
);

DROP TABLE IF EXISTS child_table;
CREATE TABLE child_table
(
  p_id INT NOT NULL,
  c_timestamp TIMESTAMPTZ NOT NULL,
  CONSTRAINT child_table_p_id_fkey FOREIGN KEY (p_id) REFERENCES parent_table ON DELETE CASCADE
);

SELECT create_hypertable('child_table', 'c_timestamp', chunk_time_interval => interval '1 day');

INSERT INTO parent_table (p_id) SELECT GENERATE_SERIES(1, 10);

INSERT INTO child_table (p_id, c_timestamp) 
SELECT FLOOR(RANDOM() * 10 + 1)::INT, GENERATE_SERIES('2020-07-01'::TIMESTAMPTZ, '2020-07-06'::TIMESTAMPTZ, '1 Day');

SELECT * FROM chunk_relation_size_pretty('child_table');
```

2. truncate the parent. This cascades to the child and removes all data, but 6 empty chunks are still present.

```
TRUNCATE parent_table CASCADE;
SELECT * FROM chunk_relation_size_pretty('child_table');
```

3. truncate the child. This will purge the chunks.

```
TRUNCATE child_table CASCADE;
SELECT * FROM chunk_relation_size_pretty('child_table');
```

**Expected behavior**
I expected the first `TRUNCATE` to delete the child's data AND purge the chunks.

**Actual behavior**
Only the child's data is purged. Empty chunks remain.

**Comment by NunoFilipeSantos** [comment]
@jflambert can you please try with our latest TimescaleDB 2.4.2 (PG 12 or PG13) and get back to us?

**Comment by jflambert** [comment]
@NunoFilipeSantos it seems to have been fixed in 2.4.2 yes (still happens in 2.4.1)

Note that `chunk_relation_size_pretty` had to be replaced with `chunks_detailed_size` or equivalent in 2.x

**Comment by NunoFilipeSantos** [comment]
Thank you @jflambert! That is great news.
